[wifi] Prevent starting AP+STA mode with correct settings

TD-er opened the pull request:
WiFi persistent settings were disabled after initialization of the event handler.
This allowed the firmware to start WiFi connection with settings stored in the flash (not our settings, the settings kept by the core library). If these fail, then a connection failed event would occur before a connect attempt was started from ESPeasy, which would lead to strange results on some nodes while others work just fine.

See #1238

TD-er committed:
[wifi] Prevent starting AP+STA mode with correct settings

WiFi persistent settings were disabled after initialisation of the event handler.
This allowed the firmware to start WiFi connection with settings stored in the flash (not our settings, the settings kept by the core library). If these fail, then a connection failed event would occur before a connect attempt was started from ESPeasy, which would lead to strange results on some nodes while others work just fine.

TD-er commented:
I've made a build to test [here](https://www.dropbox.com/s/juni70xcagp2htz/ESPeasy_build_PR1250_20180415.rar?dl=0)

TD-er commented:
OK, I've got one of the devices myself that will start with AP+STA
So, this is not a fix, but now I have something to debug.

TD-er committed:
[Wifi] force STA mode at boot

I finally have one node acting as described by others and this one reported a disconnect before a connect attempt was made.

Now forcing the wifi in STA mode, should resolve this issue.

TD-er commented:
Updated the [test build](https://www.dropbox.com/s/juni70xcagp2htz/ESPeasy_build_PR1250_20180415.rar?dl=0) I made.

TD-er committed:
[wifi] Proper log of static IP configuration

micropet commented:
Updated testbuild works also, but Webserver is always sklow.

11990 : EVENT: System#Wake
12009 : WIFI : Connecting SMC attempt #0
12024 : EVENT: System#Boot
12031 : ACT  : 
12043 : ACT  : Publish ESP-201/IP,0.0.0.0
12055 : ACT  : timerSet,1,60
12065 : ACT  : 
12089 : WD   : Uptime 0 ConnectFailures 0 FreeMem 13576
12121 : DS   : Temperature: 20.56 (28-ff-b8-ea-b4-16-3-ed)
12123 : EVENT: DS18b20#Temperature=20.56
15066 : WIFI : Connected! AP: SMC (5C:49:79:45:39:ED) Ch: 1 Duration: 3056 ms
15067 : EVENT: WiFi#ChangedAccesspoint
17943 : WIFI : DHCP IP: 192.168.0.201 (ESP-201-1) GW: 192.168.0.3 SN: 255.255.255.0   duration: 2877 ms
17954 : EVENT: WiFi#Connected
18053 : MQTT : Intentional reconnect
18125 : MQTT : Connected to broker with client ID: ESPClient_5C:CF:7F:0B:68:52
18136 : Subscribed to: ESP-201/#
18145 : EVENT: MQTT#Connected
18306 : Current Time Zone:  DST time start: 2018-03-25 02:00:00 offset: 120 minSTD time start: 2018-10-28 03:00:00 offset: 60 min
18307 : EVENT: Time#Initialized
18330 : EVENT: Clock#Time=Sun,17:24
39173 : EVENT: Clock#Time=Sun,17:25
42202 : WD   : Uptime 1 ConnectFailures 0 FreeMem 12728
72202 : WD   : Uptime 1 ConnectFailures 0 FreeMem 12728
72387 : DS   : Temperature: 20.56 (28-ff-b8-ea-b4-16-3-ed)
72389 : EVENT: DS18b20#Temperature=20.56
72464 : EVENT: Rules#Timer=1
72477 : ACT  : Publish ESP-201/IP,192.168.0.201
72497 : ACT  : Publish ESP-201/MAC,5C:CF:7F:0B:68:52
72522 : ACT  : Publish ESP-201/Time,17:25:34
72540 : ACT  : Publish ESP-201/Uptime,1
72570 : ACT  : Publish ESP-201/RSSI,-62
77585 : ACT  : Publish ESP-201/SSID,SMC
82602 : ACT  : Publish ESP-201/BSSID,5C:49:79:45:39:ED
87618 : ACT  : Publish ESP-201/CH,1
92631 : ACT  : Publish ESP-201/SYSHEAP,9032
97643 : ACT  : 
97652 : ACT  : timerSet,1,60
102676 : WD   : Uptime 2 ConnectFailures 0 FreeMem 9432
102711 : EVENT: Clock#Time=Sun,17:26
103181 : MQTT : Connection lost
103182 : EVENT: MQTT#Disconnected
103269 : MQTT : Connected to broker with client ID: ESPClient_5C:CF:7F:0B:68:52
103279 : Subscribed to: ESP-201/#
103292 : EVENT: MQTT#Connected
132676 : WD   : Uptime 2 ConnectFailures 2 FreeMem 12728
132725 : DS   : Temperature: 20.50 (28-ff-b8-ea-b4-16-3-ed)
132727 : EVENT: DS18b20#Temperature=20.50

micropet commented:
I see often Disconnects:

3083180 : MQTT : Connection lost
3083181 : EVENT: MQTT#Disconnected
3083270 : MQTT : Connected to broker with client ID: ESPClient_5C:CF:7F:0B:68:52
3083281 : Subscribed to: ESP-201/#
3083292 : EVENT: MQTT#Connected
3098935 : EVENT: Clock#Time=Sun,18:16
3107798 : WD   : Uptime 39 ConnectFailures 6 FreeMem 12560
3132935 : EVENT: Rules#Timer=1
3132945 : ACT  : Publish ESP-201/IP,192.168.0.201
3132967 : ACT  : Publish ESP-201/MAC,5C:CF:7F:0B:68:52
3132988 : ACT  : Publish ESP-201/Time,18:16:34
3133007 : ACT  : Publish ESP-201/Uptime,39
3133029 : ACT  : Publish ESP-201/RSSI,-57
3133049 : ACT  : Publish ESP-201/SSID,SMC
3138065 : ACT  : Publish ESP-201/BSSID,5C:49:79:45:39:ED
3143082 : ACT  : Publish ESP-201/CH,1
3148096 : ACT  : Publish ESP-201/SYSHEAP,8864
3153111 : ACT  : 
3153124 : ACT  : timerSet,1,60
3158149 : WD   : Uptime 40 ConnectFailures 5 FreeMem 9264
3158198 : DS   : Temperature: 20.31 (28-ff-b8-ea-b4-16-3-ed)
3158200 : EVENT: DS18b20#Temperature=20.31
3163274 : EVENT: Clock#Time=Sun,18:17
3163305 : MQTT : Connection lost
3163306 : EVENT: MQTT#Disconnected
3163387 : MQTT : Connected to broker with client ID: ESPClient_5C:CF:7F:0B:68:52
3163395 : Subscribed to: ESP-201/#
3163403 : EVENT: MQTT#Connected

TD-er commented:
The thing is you didn't see them before. Question is whether the connection was stable before, when nothing was logged.
I want to look into collecting more information about the connection like CRC errors. Not sure if they are collected somewhere.
But there is a lot more information present that can be used to diagnose the connection quality

TD-er committed:
[info] Added FW information to help diagnose issues

A lot of issue reports do not include information about the used version of the core libraries and 'normal', 'dev', etc. 
Since this information helps narrowing down the possible causes of the reported issues, it is best to add them in the general log.
